src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -654,7 +666,9 @@ ExpressionEvaluator::_evaluate_predicate_expression<ExpressionEvaluator::Bool>(
 
           // Do not handle tricky cases with lossy conversions here (e.g., `1 BETWEEN 1.1 and 2.0`).
           if (lower_bound && upper_bound) {
```

Comment:
It might be too much, but what about extracting the lower_bound casts, the checking for NULL, checking for ValueExpression, and checking for Lower and upper (`if (lower_bound && upper_bound)`) into function? Maybe taking an if lambda and an else lambda?

Reply:
I guess that makes it harder if something was done or not, i.e., if the conditions were met or not. Returning an empty vector if not sufficient because the input could be empty and we would need an additionaly returned flag to see if we need to do a default pass.

Reply:
Well, pretty much everything is doable. But you're very much into the code right now and if you say it's  not worth the hassle, let's not do it.

Reply:
We can sure still talk, but we don't need to block this PR.

Reply:
I wanted to make sure I did not break anything for TPC-H Q 14 anyway

Reply:
We could have sth like
```c++
auto result_values = pmr_vector<Bool>{};
if (_evaluate_between_expression(between_expression, result)) {
  return std::make_shared<ExpressionResult<Bool>>(std::move(result_values));
} else {
  return evaluate_expression_to_result<Bool>(*rewrite_between_expression(predicate_expression)); 
}
```
for evaluation to bitvector and alike for PosLists. Yeah, sounds reasonable

Reply:
Will wait for nemea to re-run benchmarks again.